src/main/host/syscall/socket.c

```diff
@@ -733,12 +739,6 @@ SysCallReturn syscallhandler_bind(SysCallHandler* sys,
         return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = -EINVAL};
     }
 
-    /* Make sure the addr PluginPtr is not NULL. */
-    if (!addrPtr.val) {
-        debug("binding to a NULL address is invalid");
-        return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = -EINVAL};
-    }
-
     const struct sockaddr* addr = process_getReadablePtr(sys->process, addrPtr, addrlen);
     utility_assert(addr);
```

Comment:
*this* should return -EFAULT on NULL instead of crashing. i.e. check whether accessing the pointer succeeded here, instead of whether we think it'll succeed above

Reply:
Yeah I think this is a common issue throughout the C syscall handlers. I fixed this for this case.